pkg/auth/dependency/forgotpassword/provider.go

```diff
@@ -84,12 +73,16 @@ func (p *Provider) SendCode(loginID string) (err error) {
 
 		if email {
 			err = p.sendEmail(prin.LoginID, codeStr)
-			return
+			if err != nil {
+				return
+			}
```

Comment:
I think we should return after sending even there is no error? I think creating and sending multiple code is not expected in forgot password context?